src/modals/TransactionDetailsModal.tsx

```diff
@@ -152,20 +152,29 @@ const TransactionDetailsModal = ({ transaction, address, onClose }: TransactionD
           </DetailsRow>
           <DetailsRow label={t`Inputs`}>
             <AddressList>
-              {transaction.inputs?.map((input) => (
-                <ActionLinkStyled key={`${input.outputRef.key}`} onClick={() => handleShowAddress(input.address)}>
-                  <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
-                </ActionLinkStyled>
-              ))}
+              {transaction.inputs?.map(
+                (input) =>
+                  input.address !== undefined && (
+                    <ActionLinkStyled
+                      key={`${input.outputRef.key}`}
+                      onClick={() => handleShowAddress(input.address ?? '')}
+                    >
+                      <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
+                    </ActionLinkStyled>
+                  )
+              )}
             </AddressList>
           </DetailsRow>
           <DetailsRow label={t`Outputs`}>
             <AddressList>
-              {transaction.outputs?.map((output) => (
-                <ActionLinkStyled key={`${output.key}`} onClick={() => handleShowAddress(output.address)}>
-                  <AddressEllipsed key={`${output.key}`} addressHash={output.address} />
-                </ActionLinkStyled>
-              ))}
+              {transaction.outputs?.map(
+                (output) =>
+                  output.address && (
+                    <ActionLinkStyled key={`${output.key}`} onClick={() => handleShowAddress(output.address)}>
+                      <AddressEllipsed key={`${output.key}`} addressHash={output.address} />
+                    </ActionLinkStyled>
+                  )
+              )}
```

Comment:
There is no need to check for `output.address`. It's never `undefined`. The following is enough:

```ts
{transaction.outputs?.map((output) => (
                <ActionLinkStyled key={`${output.key}`} onClick={() => handleShowAddress(output.address)}>
                  <AddressEllipsed key={`${output.key}`} addressHash={output.address} />
                </ActionLinkStyled>
              ))}
```